### DISCUSSION TITLE
Can web browsers read JSX directly?

### DISCUSSION THREAD

**Discussion by 0xprincedev** [community/community#20753] [Discussions]
Web browsers cannot read JSX directly. This is because they are built to only read regular JS objects and JSX is not a regular JavaScript object
 
For a web browser to read a JSX file, the file needs to be transformed into a regular JavaScript object. For this, we use Babel

**Comment by The-Real-Thisas** [comment]
> Browsers cannot read JSX because there is no inherent implementation for the browser engines to read and understand it. You can use babel to transform your jsx into native javascript and HTML which browser can understand.

From: https://stackoverflow.com/a/54663149/13249661

Couldn't have said it better myself.

Hope this helps,
Thisas

PS: @BlueDream0807 if you were attempting to answer your own question, I don't think thats currently possible though I might be wrong. Glad you were able to figure it out though.
